tests/unittest/_torch/thop/parallel_hw_agnostic/test_minimax_m3_fp8_indexer.py

```
@@ -0,0 +1,314 @@
+            rotary_dim=rotary_dim,
+        )
+
+
+def test_minimax_m3_fp8_indexer_rejects_bad_cache_contracts() -> None:
+    """Cache dtype, rank, and slot-vector length are validated before launch."""
+    qk = torch.empty(2, 5 * 128, dtype=torch.bfloat16, device="cuda")
+    slots = torch.zeros(2, dtype=torch.int32, device="cuda")
+    weights = torch.ones(128, dtype=torch.bfloat16, device="cuda")
+    positions = torch.zeros(2, dtype=torch.int32, device="cuda")
+
+    with pytest.raises(RuntimeError, match="must use torch.float8_e4m3fn"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)